src/ui/notify.c

```diff
@@ -442,9 +442,10 @@ static GApplication *application = NULL;
 
 void sc_notify_init(void)
 {
-	sc_notify_close();
-	application = g_application_new("org.opensc.notify", G_APPLICATION_NON_UNIQUE);
-	if (application) {
+	if (!application) {
+		application = g_application_new("org.opensc.notify", G_APPLICATION_NON_UNIQUE);
+	}
+	if (application && FALSE == g_application_get_is_registered(application)) {
```

Comment:
`g_application_new` isn't supposed to return `NULL` (at least the docs don't state `NULL` as a possible return value)
```suggestion
	if (FALSE == g_application_get_is_registered(application)) {
```

Reply:
I can't imagine how `g_application_new()` could *always* allocate enough memory. And even if that would be possible, checking for `application != NULL` doesn't hurt.